app/src/main/java/com/github/multimatum_team/multimatum/model/DeadlineAdapter.kt

```diff
@@ -10,19 +10,37 @@ import android.view.ViewGroup
 import android.widget.BaseAdapter
 import android.widget.TextView
 import com.github.multimatum_team.multimatum.R
+import com.github.multimatum_team.multimatum.service.ClockService
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import java.time.Period
+import javax.inject.Inject
+
+@EntryPoint
+@InstallIn(SingletonComponent::class)
+interface ClockServiceEntryPoint {
+    fun provideClockService(): ClockService
+}
 
 /**
 Class which is used to show the deadline in a clear list.
 Based on the tutorial:
 https://www.raywenderlich.com/155-android-listview-tutorial-with-kotlin
  */
-class DeadlineAdapter(private val context: Context) : BaseAdapter() {
+class DeadlineAdapter @Inject constructor(private val context: Context) : BaseAdapter() {
```

Comment:
I didn't see in the code somewhere where this injection is used. Did I just miss it ?

Reply:
Good catch, that was from an earlier attempt to inject the clock service. It seems to work without it.
Though dependency injection _is_ used in this class (the clockService is injected). Since Hilt doesn't support injecting dependencies into adapters (it only supports a few android classes out of the box), I have to define an entry point and maually inject the clock service.